danceradvent/public/articles/2023/15-mason-whats-old-is-new-again.pod

```diff
@@ -0,0 +1,133 @@
+=head1 Dancer2::Plugin::Mason - What's old is new again!
+
+This year, Retro Santa paid us a visit and gave us a new gift of something
+old - an L<HTML::Mason> plugin for Dancer2!
+
+=head2 But Santa, why? WHY?
+
+Mason is a solid templating engine, but it lacks a controller. A
+number of projects have built their own controllers (such as
+L<Request Tracker (RT)|https://bestpractical.com/request-tracker>), and
+others struggle without one. That's where Dancer2 comes in.
+
+L<Dancer2::Template::Mason> is a template engine for HTML::Mason. If
+HTML::Mason is your favorite template engine, you can now use it
+in your favorite web framework. If you have legacy Mason applications
+that would benefit from having a proper controller, or a general
+refresh with some a more modern design, Dancer2 has your back.
+
+=head2 Does this mean I'm NOT on the Naughty List?
+
+Not at all! It's actually a pretty good gift, combining two powerful tools
+that are already in the toolbox of a lot of Perl developers. Go ahead,
+you can thank Retro Santa now. I'll wait.
+
+HTML::Mason, like all frameworks, has its warts and annoyances, but also
+a lot of upside, and not having HTML::Mason as part of the Dancer2
+ecosystem has been a glaring oversight. No more!
+
+=head2 How do I use it?
+
+First, make your layout (F<views/layouts/main.m>):
+
+    <html lang="en">
+        <head>
+            <title><% $title %></title>
+        </head>
+
+        <body>
+            <% $content %>
+        </body>
+    </html>
+
+    <%args>
+    $title => "It's HTML::Mason... in Dancer2!"
+    $content
+    $deferred
+    $perl_version
+    $dancer_version
+    $settings
+    $request
+    $params
+    $vars
+    $session
+    </%args>
+
+That pile of variables in C<%args>? Like other template engines, Dancer2
+gives you that automatically. HTML::Mason, unlike some other template
+engines (L<Template::Toolkit> for example) makes you explicitly declare
+variables before using them.
+
+Too many variables? You can always reference them via the C<%ARGS> hash.
+
+Next, create your page template (F<views/index.m>):
+
+    <p>Hello, <% $ARGS{ name } %>!</p>
+
+And finally, add a route that uses your Mason template:
+
+    get '/:name?' => sub {
+        my $name = body_paramaters->get( 'name' ) // 'Mystery Visitor';
+        template 'index', { name => $name };
+    };
+
+and Bob's your uncle! Now, when you go to C<http://localhost:5000/Uncle%20Bob>
+you'll see:
+
+    Hello, Uncle Bob!
+
+And if you go to C<http://localhost/:5000> you get:
+
+    Hello, Mystery Visitor!
+
+=head2 Care and feeding of your Mason templates
```

Comment:
This section has the problem that you apologize for both sides of the config. :-) You begin with "to get the best performance you have to do some work", and "to develop without going nutso you have to change the config". 

Turn that around! The default config is going to work beautifully with the dev process, and when the time comes to go to prod, you only have to change one line of config and VA-VA-VOOOOM!

Reply:
This makes perfect sense.